libs/blocks/preflight/panels/assets.js

```diff
@@ -0,0 +1,188 @@
+import { html, signal, useEffect } from '../../../deps/htm-preact.js';
+import { createTag } from '../../../utils/utils.js';
+
+function isViewportTooSmall() {
+  return !window.matchMedia('(min-width: 1200px)').matches;
+}
+
+const imagesWithMismatch = signal([]);
+const imagesWithMatch = signal([]);
+const checksPerformed = signal(false);
+const viewportTooSmall = signal(isViewportTooSmall());
+
+const groups = [
+  { title: 'Images with dimension mismatch', imgArray: imagesWithMismatch },
+  { title: 'Images with matching dimensions', imgArray: imagesWithMatch },
+];
+
+async function checkImageDimensions() {
+  if (checksPerformed.value || viewportTooSmall.value) return;
+  checksPerformed.value = true;
+
+  const allImages = [...document.querySelectorAll('main picture img')];
+  if (!allImages.length) return;
+
+  // Force load all images
+  await Promise.all(
+    allImages.map((img) => {
+      if (!img.complete) {
+        img.setAttribute('loading', 'eager');
+        return new Promise((resolve) => {
```

Comment:
suggestion
```js
return new Promise(resolve => img.addEventListener('load', resolve));
```

Reply:
This leads to an ESLint error `Return values from promise executor functions cannot be read`, so it's more straightforward to use the current option.